### ISSUE TITLE
Yul compilation failed with Error: ParserError: Expected 'StringLiteral' but got 'ILLEGAL'

### ISSUE THREAD

**Issue by agroce** [ethereum/fe#507]
Compiling the attached file (attached since contains a character not handled here) produces:

[si.zip](https://github.com/ethereum/fe/files/6962971/si.zip)

```
Error: ParserError: Expected 'StringLiteral' but got 'ILLEGAL'
   --> input.yul:524:83:
    |
524 |         data "0x5c179d3bfde4c521afc3d3944357db5ee881a69c237d67c9aa79aa7a027c40ea" ""
    |                                                                                   ^^


thread 'main' panicked at 'Yul compilation failed with the above errors', crates/driver/src/lib.rs:86:17
stack backtrace:
   0:     0x55715b0ff270 - std::backtrace_rs::backtrace::libunwind::trace::h63b7a90188ab5fb3
                               at /rustc/9bc8c42bb2f19e745a63f3445f1ac248fb015e53/library/std/src/../../backtrace/src/backtrace/libunwind.rs:90:5
   1:     0x55715b0ff270 - std::backtrace_rs::backtrace::trace_unsynchronized::h80aefbf9b851eca7
                               at /rustc/9bc8c42bb2f19e745a63f3445f1ac248fb015e53/library/std/src/../../backtrace/src/backtrace/mod.rs:66:5
   2:     0x55715b0ff270 - std::sys_common::backtrace::_print_fmt::hbef05ae4237a4d72
                               at /rustc/9bc8c42bb2f19e745a63f3445f1ac248fb015e53/library/std/src/sys_common/backtrace.rs:67:5
   3:     0x55715b0ff270 - <std::sys_common::backtrace::_print::DisplayBacktrace as core::fmt::Display>::fmt::h28abce2fdb9884c2
                               at /rustc/9bc8c42bb2f19e745a63f3445f1ac248fb015e53/library/std/src/sys_common/backtrace.rs:46:22
   4:     0x55715b11845f - core::fmt::write::h3b84512577ca38a8
                               at /rustc/9bc8c42bb2f19e745a63f3445f1ac248fb015e53/library/core/src/fmt/mod.rs:1092:17
   5:     0x55715b0fccd2 - std::io::Write::write_fmt::h465f8feea02e2aa1
                               at /rustc/9bc8c42bb2f19e745a63f3445f1ac248fb015e53/library/std/src/io/mod.rs:1572:15
   6:     0x55715b101405 - std::sys_common::backtrace::_print::h525280ee0d29bdde
                               at /rustc/9bc8c42bb2f19e745a63f3445f1ac248fb015e53/library/std/src/sys_common/backtrace.rs:49:5
   7:     0x55715b101405 - std::sys_common::backtrace::print::h1f0f5b9f3ef8fb78
                               at /rustc/9bc8c42bb2f19e745a63f3445f1ac248fb015e53/library/std/src/sys_common/backtrace.rs:36:9
   8:     0x55715b101405 - std::panicking::default_hook::{{closure}}::ha5838f6faa4a5a8f
                               at /rustc/9bc8c42bb2f19e745a63f3445f1ac248fb015e53/library/std/src/panicking.rs:208:50
   9:     0x55715b100eb3 - std::panicking::default_hook::hfb9fe98acb0dcb3b
                               at /rustc/9bc8c42bb2f19e745a63f3445f1ac248fb015e53/library/std/src/panicking.rs:225:9
  10:     0x55715b04bdc6 - fe_common::panic::DEFAULT_PANIC_HOOK::{{closure}}::{{closure}}::h0c65fa1148ba1b12

You've hit an internal compiler error. This is a bug in the Fe compiler.
Fe is still under heavy development, and isn't yet ready for production use.

If you would, please report this bug at the following URL:
  https://github.com/ethereum/fe/issues/new
Aborted
```

Using

```
[profile.dev]
debug = true
panic = "abort"
```

for fuzzing.  Using https://github.com/agroce/afl-compiler-fuzzer

**Comment by cburgdorf** [comment]
What a coincidence #506 is going to fix this (and a few other cases)

**Comment by agroce** [comment]
I need to dome some fixups on my triage corpus to adjust to the "def --> fn" thing which really had me confused a while this morning, but I'll wait until github is working better :)

**Comment by cburgdorf** [comment]
Yeah, there are a few syntax changes that we want to get in before we cut a beta version later this year. Another one that is coming shortly is the introduction of `let` for variable declarations: https://github.com/ethereum/fe/pull/509

**Comment by agroce** [comment]
@cburgdorf This is fixed post merge of #509 and can be closed.

There is a new "variant" I just submitted (#551) though, of not checking for illegal string contents.
